src/cmap/connection.ts

```diff
@@ -170,7 +170,9 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
   address: string;
   socketTimeoutMS: number;
   monitorCommands: boolean;
+  /** Indicates that the connection (including underlying TCP socket) has been closed. */
   closed: boolean;
+  /** Indicates that the connection has been explicitly destroyed. In the destroyed state, the connection is also closed */
   destroyed: boolean;
```

Comment:
Not a request for changes or anything, but I don’t quite understand how we’re semantically making a difference between `.closed` and `.destroyed`. I understand that `.destroyed` is the result of an explicit call to `.destroy()`, but in which case would we check for that instead of using `.closed`?

Combining this with @baileympearson’s comment immediately below, it seems to me a bit like ideally we would:
- Always call `this.destroy()` in the places where we destroy _either_ `this[kStream]` or `this[kMessageStream]` in order to clean up whatever has not been cleaned up yet
- Drop the `.destroyed` property entirely
- Maybe go even further and just make `.closed` a getter that returns `this[kStream].destroyed && this[kMessageStream].destroyed` or similar, to avoid having to track separate state?